### PR TITLE
feat(tooltip): add tooltip component Al 2049

### PR DESCRIPTION
**Pull request by ianaa**
the storybook examples contain a pattern for icon buttons (most common in our apps).
@manuatarchilogic @romandkos @luk-zap let me know if you think this is good enough/convenient enough or would you much rather prefer a directive? 
note that I will be adding  a `shortcuts` prop to tooltips in a follow-up PR a bit later on along with a keyboard shortcut component (to be moved from editor repo)

### CODE REVIEW

File: src/components/__tests__/Tooltip.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  slots: {
+    default: `
+          <template #default="slotProps">
+            <button v-bind="slotProps">hi</button>
+          </template>`
+  }
+}
+
+describe('Tooltip.vue', () => {
+  describe('then the tooltip role is "label"', () => {
```
Comment (manuatarchilogic) [inline src/components/__tests__/Tooltip.test.ts]:
```suggestion
  describe('when the tooltip role is "label"', () => {
```

File: src/components/__tests__/Tooltip.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    it('assigns an "id" to the tooltip element and "aria-labelledby" to the control', () => {
+      const { getByRole } = render(Tooltip, { ...tooltipPropsAndSlots })
+      expect(getByRole('button')).toHaveAttribute(
+        'aria-labelledBy',
+        expect.stringContaining(tooltipId)
+      )
+      expect(getByRole('tooltip')).toHaveAttribute('id', expect.stringContaining(tooltipId))
+    })
+  })
+  describe('then the tooltip role is "description"', () => {
```
Comment (manuatarchilogic) [inline src/components/__tests__/Tooltip.test.ts]:
```suggestion
  describe('when the tooltip role is "description"', () => {
```

File: src/stories/Tooltip.stories.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import type { Meta, StoryObj } from '@storybook/vue3'
+import type { ComponentProps } from 'vue-component-type-helpers'
+import { ATooltip, AButton } from '../components'
+import { userEvent, waitFor, within } from '@storybook/testing-library'
+
+type TooltipProps = ComponentProps<typeof ATooltip>
+
+/**
+ * A tooltip component provides extra information about a control than triggers it.
```
Comment (manuatarchilogic) [inline src/stories/Tooltip.stories.ts]:
```suggestion
 * A tooltip component provides extra information about the control that triggers it.
```

File: src/stories/Tooltip.stories.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  await waitFor(async () => {
+    await userEvent.hover(button)
+  })
+}
+
+/**
+ * Below are examples of differently positioned tooltips
+ * based on available space on various sides of the screen.
+ *
+ * The positioning logic is handled by [Floating UI](https://floating-ui.com/docs/vue#options) library,
```
Comment (manuatarchilogic) [inline src/stories/Tooltip.stories.ts]:
```suggestion
 * The positioning logic is handled by the [Floating UI](https://floating-ui.com/docs/vue#options) library,
```

File: src/stories/Tooltip.stories.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+ * with the use of `flip()` and `shift()` [middleware](https://floating-ui.com/docs/middleware).
+ *
+ * If there's available space on every side of the tooltip,
+ * it is rendered on the top center of the
```
Comment (manuatarchilogic) [inline src/stories/Tooltip.stories.ts]:
```suggestion
 * it is rendered on the top center of the control
```

**Review by manuatarchilogic — APPROVED** [review]
nice!

just noticed that Dashboard uses `"v-tooltip": "^4.0.0-beta.2"` and Editor `"floating-vue": "^2.0.0-beta.24"` 🙈

File: src/stories/Tooltip.stories.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+ * or with an ESC key while the control is focused.
+ *
+ * Tooltips are positioned automatically and re-positioned if the content is scrolled,
+ * taking into account available space (handled by [Floating UI](https://floating-ui.com/docs/getting-started)).
+ */
+const meta: Meta<TooltipProps> = {
+  component: ATooltip,
+  title: 'Components/Tooltip',
+  args: {
+    content: 'Peekaboo!'
```
Comment (romiekos) [inline src/stories/Tooltip.stories.ts:28]: It is suitable! 😄

File: src/components/Tooltip.vue [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      :style="floatingStyles"
+      class="absolute top-0 left-0 max-w-[160px] min-w-10 z-10"
+      :class="isVisible ? 'visible' : 'invisible'">
+      <div
+        ref="floatingArrow"
+        class="absolute w-2 h-2 bg-white rotate-45"
+        :style="arrowStyle"></div>
+      <div
+        class="flex items-center justify-center body-sm text-stone rounded shadow-sm bg-white py-1 px-2 m-2 gap-1">
+        <span v-if="content" class="py-[2px]">{{ content }}</span>
```
Comment (romiekos) [inline src/components/Tooltip.vue:122]: It is not possible (not needed) to add a custom HTML inside tooltips?

Comment (ianaa) [reply src/components/Tooltip.vue:122]: yes, it's intentional. to ensure we don't add any buttons/links/interactive elements

**Review by romiekos — APPROVED** [review]
👍

**Pull request merged**